### ISSUE TITLE
Is there a specific reason you chose RepMixer blocks over alternatives like Swin Transformer or standard CNN blocks?

### ISSUE THREAD

**Issue by Mohamed-Dhouib** [apple/ml-fastvlm#18]
Hello,
Could you please clarify why you chose to use RepMixer blocks instead of other possible alternatives like Swin Transformer or ConvNeXt blocks? Did you conduct any ablation studies to support this choice? Thank you!

**Comment by pavank-apple** [comment]
@Mohamed-Dhouib 
Our choice was based on the findings of [FastViT](https://arxiv.org/abs/2303.14189) paper, we primarily focussed on lowering the latency of the encoder for high resolution images. We evaluated the performance of our image encoder against ConvNeXt variants that are CLIP pre-trained, as shown in Figure 1 and Table 4 of the paper. Our results indicate that FastViTHD offers the best accuracy-latency tradeoff. We were unable to find any CLIP pre-trained Swin Transformer models, which is why they are not included in the comparisons presented in Table 4.

**Comment by Mohamed-Dhouib** [comment]
@pavank-apple Thanks for the answer.
Actually, we can't draw conclusions about the pertinence of using RepMixer based on a simple comparison with ConvNeXt, since the superiority of your proposed encoder may stem from its hierarchical design, regardless of the specific blocks used.
It seems to me that RepMixer may not necessarily be better than using ConvNeXt blocks or Swin Transformer blocks.
Also, your encoder wasn't CLIP-pretrained, right?
Anyway, great work! It would be great if you could provide code to reproduce the training of your model. This way, more people can build on your work.
Honestly, so much can go wrong when trying to reproduce VLM pretraining.

**Comment by pavank-apple** [comment]
@Mohamed-Dhouib 
I agree with your comments, RepMixer is an extremely simple block, its literally a DW conv layer at inference time, it will be inferior to shifted window attention blocks in Swin transformer. As I had mentioned earlier, our choice was purely motivated by inference latency. Our encoder is CLIP pre-trained, other pretrainings tend to be inferior when used in a VLM setup (as shown in [Cambrian-1](https://arxiv.org/abs/2406.16860) paper.) We have provided all the code to reproduce training, all of the training logic can be found [here](https://github.com/apple/ml-fastvlm/blob/main/llava/train/train_qwen.py). You can simply reuse the scripts from LLaVA repo to launch your training runs. The only thing missing at the moment is the CLIP pre-trained FastViTHD checkpoint, which we are working on publishing soon.

**Comment by lawsonxwl** [comment]
@pavank-apple  
Thanks for sharing the pretrained VLM model! 
But I've found that open-sourced FastViTHD model is a reparameterized one, I wonder if I can train this vision encoder along with LLMs directly? Will VLM training over a reparameterized vision model hurt the final performance of the downstream tasks? If so, could you open-source the raw vision encoder weights before reparameterization?
